pennylane/measurements/measurements.py

```diff
@@ -200,32 +200,27 @@ def numeric_type(self):
                 unrecognized and cannot deduce the numeric type
         """
         raise qml.QuantumFunctionError(
-            "Cannot deduce the numeric type of the measurement process with unrecognized "
-            + f"return_type {self.return_type}."
+            f"The numeric type of the measurement {self.__class__.__name__} is not defined."
         )
 
     def shape(self, device=None):
         """The expected output shape of the MeasurementProcess.
 
-        Note that the output shape is dependent on the device when:
-
-        * The ``return_type`` is either ``Probability``, ``State`` (from :func:`.state`) or
-          ``Sample``;
-        * The shot vector was defined in the device.
+        Note that the output shape is dependent on the device when the shot vector is defined in
+        the device.
```

Comment:
I feel like we should mention the first case as an option here, to let developers know that the measurement process's shape can depend on the device through other means, not just through the shot vector.